### ISSUE TITLE
Gzip encoding error when using httpclient adapter

### ISSUE THREAD

**Issue by milgner** [zendesk/zendesk_api_client_rb#278]
Since we use the `httpclient` gem in a lot of places, I also set `config.adapter = :httpclient` for the Zendesk API client. This however leads to the following error in each request:

```
get https://evopark.zendesk.com/api/v2/users/search?external_id=8
  Accept: "application/json"
  Accept-Encoding: "gzip;q=1.0,deflate;q=0.6,identity;q=0.3"
  User-Agent: "ZendeskAPI Ruby 1.12.1"
Zlib::GzipFile::Error: not in gzip format
from /home/marcus/.rvm/gems/ruby-2.1.7@evopark/gems/zendesk_api-1.12.1/lib/zendesk_api/middleware/response/gzip.rb:13:in `initialize'
```

I can however confirm that a curl request with `-H "Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3"` piped through `gzip -d` works.

Also, not specifying the adapter (and thus defaulting to `Net::HTTP`) works, too.

**Comment by steved** [comment]
Does httpclient handle gzip decoding itself? I can add the exclusion here: https://github.com/zendesk/zendesk_api_client_rb/blob/master/lib/zendesk_api/client.rb#L150

**Comment by milgner** [comment]
Thanks for the quick feedback. Apparently httpclient supports it since https://github.com/nahi/httpclient/issues/42

**Comment by steved** [comment]
I've released v1.13.1 with the exception added. Let me know if there are other problems.
